soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <soc.h>
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+void z_arm_platform_init(void)
+{
+	am_hal_pwrctrl_low_power_init();
+	am_hal_pwrctrl_control(AM_HAL_PWRCTRL_CONTROL_SIMOBUCK_INIT, 0);
+	am_hal_rtc_osc_disable();
+}
```

Comment:
question: is there any reason to not use a SYS_INIT as many other SoCs do?

Reply:
Added `SYS_INIT`.